### PR TITLE
builder/oracle: Use official OCI Go SDK

### PR DESCRIPTION
**Pull request by harveylowndes**
This change migrates the Oracle builder to use the official OCI Go SDK and removes the hand crafted client.

### CODE REVIEW

File: builder/oracle/oci/driver_oci.go [hunk 1]
```diff
@@ -1,123 +1,184 @@
...
 		id,
 		waitStates,
 		terminalState,
+		0,             //Unlimited Retries
+		5*time.Second, //5 second wait between retries
 	)
 }
+
+// WaitForResourceToReachState checks the response of a request through a polled get and waits until the desired state or until the max retried has been reached.
+func waitForResourceToReachState(GetResourceState func(string) (string, error), id string, waitStates []string, terminalState string, maxRetries int, waitDuration time.Duration) error {
```
Comment (prydie) [inline builder/oracle/oci/driver_oci.go]: `s/GetResourceState/getResourceState/`

File: builder/oracle/oci/driver_oci.go [hunk 2]
```diff
@@ -1,123 +1,184 @@
...
+		}
+
+		if stringSliceContains(waitStates, state) {
+			time.Sleep(waitDuration)
+			continue
+		} else if state == terminalState {
+			return nil
+		}
+
+		return fmt.Errorf("Unexpected resource state %s, expecting a waiting state %s or terminal state  %s ", state, waitStates, terminalState)
```
Comment (prydie) [inline builder/oracle/oci/driver_oci.go]:
```go
return fmt.Errorf("unexpected resource state %q: expecting a waiting state %q or terminal state %q", state, waitStates, terminalState) 
```

File: builder/oracle/oci/driver_oci.go [hunk 3]
```diff
@@ -1,123 +1,184 @@
...
+	}
+
+	return fmt.Errorf("Maximum number of retries (%d) exceeded; resource did not reach state %s", maxRetries, terminalState)
```
Comment (prydie) [inline builder/oracle/oci/driver_oci.go]:
```go
return fmt.Errorf("maximum number of retries (%d) exceeded: resource did not reach state %q", maxRetries, terminalState)
```

File: builder/oracle/oci/driver_oci.go [hunk 4]
```diff
@@ -1,123 +1,184 @@
...
+}
+
+// stringSliceContains loops through a slice of strings returning a boolean based on whether a given value is contained in the slice.
```
Comment (prydie) [inline builder/oracle/oci/driver_oci.go]: Tend to wrap doc strings at 80 chars for readability in Go.

File: builder/oracle/oci/driver_oci.go [hunk 5]
```diff
@@ -1,123 +1,184 @@
...
+		},
 		id,
 		waitStates,
 		terminalState,
+		0,             //Unlimited Retries
+		5*time.Second, //5 second wait between retries
 	)
 }
+
+// WaitForResourceToReachState checks the response of a request through a polled get and waits until the desired state or until the max retried has been reached.
```
Comment (prydie) [inline builder/oracle/oci/driver_oci.go]: Tend to wrap doc strings at 80 chars for readability in Go.

File: builder/oracle/oci/step_image.go [hunk 6]
```diff
@@ -27,7 +27,8 @@ func (s *stepImage) Run(_ context.Context, state multistep.StateBag) multistep.S
 		return multistep.ActionHalt
 	}
 
-	err = driver.WaitForImageCreation(image.ID)
+	err = driver.WaitForImageCreation(*image.Id)
+
```
Comment (prydie) [inline builder/oracle/oci/step_image.go]: ~~`\n`~~

**Comment by prydie** [comment]
@harveylowndes Additionally a dependency on https://github.com/subosito/gotenv appears to have snuck in to the PR that I don't believe is required.

File: vendor/vendor.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
 	"comment": "",
-	"ignore": "test appengine",
```
Comment (mwhooker) [inline vendor/vendor.json]: was this intentional?

Comment (prydie) [reply vendor/vendor.json]: I wouldn't have thought so. @harveylowndes can you remove it please?

**Pull request merged**